### ISSUE TITLE
Uniqueness typing

### ISSUE THREAD

**Issue by griwes** [reaver-project/vapor#30]
Implement the `unique` specifier, to prepare for future work on disallowing unsynchronized shared access to mutable objects.

This is only meaningful on references and reference types really, so depends on #24.

* [ ] RFC and specification
* [ ] parsing
* [ ] analysis
* [ ] codegen
* [ ] tests
